packages/core/src/analytics/integrations/Integration.js

```diff
@@ -19,16 +20,58 @@ class Integration {
     this.strippedDownAnalytics = strippedDownAnalytics;
   }
 
+  /**
+   * "consentCategories" that each integration can specify accordingly.
+   */
+  static [CONSENT_CATEGORIES_PROPERTY] = null;
+
+  /**
+   * Sets the given consent on the Class static property, so it can be accessed before the instance gets created.
+   *
+   * @param {object} [options] - Integration options.
+   */
+  static setConsentCategories(options) {
+    const consentCategories = options?.[CONSENT_CATEGORIES_PROPERTY];
+
+    if (!consentCategories) {
+      return;
+    }
+
+    if (
+      typeof consentCategories !== 'string' &&
+      !Array.isArray(consentCategories)
+    ) {
+      throw new TypeError(
+        `[${this.prototype.constructor.name}] - "${CONSENT_CATEGORIES_PROPERTY}" is not an array nor a string. Make sure you are passing a valid consent type.`,
+      );
+    }
+
+    this[CONSENT_CATEGORIES_PROPERTY] = options?.[CONSENT_CATEGORIES_PROPERTY];
+  }
+
   /**
    * Method to check if the integration is ready to be loaded.
+   * It will set any given categories if provided via the options before proceeding with the check.
    *
-   * @param {object} consent - The current consent given by the user.
+   * @param {object} consented - The current consent given by the user.
+   * @param {object} options   - Integration options.
    *
    * @returns {boolean} If the integration is ready to be loaded.
    */
-  // eslint-disable-next-line no-unused-vars
-  static shouldLoad(consent) {
-    return false;
+  static shouldLoad(consented = {}, options) {
+    this.setConsentCategories(options);
```

Comment:
I am not sure if we should set the consent categories in a static property of the class. Imagine a scenario where I might want 2 instances of the same integration to be loaded with different consent categories. With the static property, the last one whose `shouldLoad` method was run would override the value for all instances.

I know this might be a very peculiar scenario but we should keep in mind that our api allows users to do this:

```js
analytics.addIntegration('int1', MyIntegration, {consentCategories: ['a', 'b']});
analytics.addIntegration('int2', MyIntegration, {consentCategories: ['c', 'd']});
```

If this is valid, then we should not store values in static properties which contain information for each integration instance that was added.

I would suggest removing the `CONSENT_CATEGORIES_PROPERTY` from the `Integration` class and having this instead:

- Add a `defaultConsentCategories` (or `getDefaultConsentCategories` method) to the Integration class which subclasses can override with the default value of the consent categories that will make the integration load.
- In `shouldLoad` do not store the consentCategories from options and use it just as an override of the `defaultConsentCategories`. I believe it is not necessary to store them since it is stored inside the options when `addIntegration` was called.
- Change `setConsentCategories` to `validateConsentCategories` which would only validate if the integration has defined the consent categories for the integration correctly (either via `defaultConsentCategories` or `options.consentCategories`) and it does not need to be inside the Integration class (For instance, a custom integration could override that method, which I guess it shouldn't?)

What do you think? This is a bit theoretical, if you want we can discuss this offline.

Reply:
As discussed offline, changed the part where we were overriding the default consent category when we receive a different one from the options - it will only be used for validation and within the `shouldLoad` method to make the decision to load (or not) the integration.
Kept the `consentCategories` naming for "ergonomic" reasons, this way we can use the same key everywhere, keeping the initial goal of not having too much things referring "custom" vs "default" consent.

Let me know if this is okay for you and thanks for the feedback 🙂

Reply:
Okay, this is fine. Thanks!